solidity/contracts/ConnextVestingWallet.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import {VestingWallet, VestingWalletWithCliff} from './VestingWalletWithCliff.sol';
+
+import {Ownable} from '@openzeppelin/contracts/access/Ownable.sol';
+import {Ownable2Step} from '@openzeppelin/contracts/access/Ownable2Step.sol';
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+/**
+ * @dev Vesting schedule:
+ *      - 1/13 of the tokens will be released after 1 year, starting from _vestingStartDate
+ *      - 1/13 of the tokens will be released every month after that, for 12 months
+ *
+ *     The equivalent vesting schedule has a 13 months duration, with a 1 month cliff, offsetted to
+ *     start from `Sept 5th 2024 - 1 month`: At Sept 5th 2024 the cliff is triggered unlocking
+ *     1/13 of the tokens, and then 1/13 of the tokens will be linearly unlocked every month after that.
+ */
+contract ConnextVestingWallet is VestingWalletWithCliff, Ownable2Step {
+  address public paymentToken;
+  uint64 public initTimestamp;
+
+  uint64 public constant ONE_YEAR = 365 days;
+  uint64 public constant ONE_MONTH = ONE_YEAR / 12;
+
+  uint64 public constant VESTING_OFFSET = ONE_YEAR - ONE_MONTH;
+  uint64 public constant VESTING_DURATION = ONE_YEAR + ONE_MONTH;
+  uint64 public constant VESTING_CLIFF_DURATION = ONE_MONTH;
+  uint256 public constant TOTAL_AMOUNT = 24_960_000 ether;
```

Comment:
if we're making timestamp, paymentToken (i don't think this one is necessary) and beneficiary settable, i believe we should also make totalAmount settable as well
what @0xGorilla said is that these contracts may be used by other connext partners (that's why i don't think NEXT should be settable) to make other payments